presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -166,7 +166,7 @@
     private final Visitor visitor;
 
     // identity-based cache for LIKE expressions with constant pattern and escape char
-    private final IdentityHashMap<LikePredicate, Regex> likePatternCache = new IdentityHashMap<>();
+    private final IdentityHashMap<LikePredicate, RegexWrapper> likePatternCache = new IdentityHashMap<>();
```

Comment:
Can't quite get the rationale behind this patch:
- `Regex` is on the value side, why hashcode function is necessary for something on the value?
- Even for the key side, it is using an `IdentityHashMap`, the native identity hash is used to generate the hash key for `LikePredicate`.

Reply:
No its not this cache its the predicateCache in RowExpressionPredicateCompiler

Reply:
The CacheKey is RowExpression so for a LikePattern it becomes a CallExpression with a ConstantExpression with value as a Regex